pkg/webui/components/location-form/index.js

```diff
@@ -0,0 +1,188 @@
+// Copyright © 2019 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React, { Component } from 'react'
+import { defineMessages } from 'react-intl'
+import * as Yup from 'yup'
+import bind from 'autobind-decorator'
+
+import Form from '../form'
+import Message from '../../lib/components/message'
+import Field from '../field'
+import SubmitBar from '../submit-bar'
+import Notification from '../notification'
+import ModalButton from '../button/modal-button'
+import Button from '../button'
+import toast from '../toast'
+
+import sharedMessages from '../../lib/shared-messages'
+import PropTypes from '../../lib/prop-types'
+
+import {
+  latitude as latitudeRegexp,
+  longitude as longitudeRegexp,
+  int32 as int32Regexp,
+} from '../../console/lib/regexp'
```

Comment:
you can define these within the file, we will be able to reuse them later for gateways as well. no need to import console related stuff to components.

Reply:
This is one reason why I would like to have a way to define components that are composed in the context of an app (in this case the `console`). Basically, something that is one hierarchy level under `views`.

Reply:
What we can do is create additional `/console/components` folder with console specific components

Reply:
Sounds good. I'll open an issue for this and then we can do it properly for all components. For this PR, I'll refactor as you said then.